tests/regression/py_bindings/implementation.cpp

```diff
@@ -0,0 +1,63 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2019, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include <utility>
+
+#include <pybind11/pybind11.h>
+
+#include <gridtools/sid/concept.hpp>
+#include <gridtools/stencil/cartesian.hpp>
+#include <gridtools/stencil/global_parameter.hpp>
+#include <gridtools/stencil/naive.hpp>
+#include <gridtools/storage/adapter/python_sid_adapter.hpp>
+
+namespace py = pybind11;
+
+using namespace gridtools;
+using namespace stencil;
+using namespace cartesian;
+
+struct copy_functor {
+    using in = in_accessor<0>;
+    using out = inout_accessor<1>;
+    using param_list = make_param_list<in, out>;
+    template <class Eval>
+    GT_FUNCTION static void apply(Eval &&eval) {
+        eval(out()) = eval(in());
+    }
+};
+
+// Here is a generic implementation of the copy algorithm.
+// The input and output fields are passed as SIDs
+template <class From, class To>
+void copy(From &&from, To &&to) {
+    static_assert(is_sid<From>(), "");
+    static_assert(is_sid<To>(), "");
+    auto &&size = sid::get_upper_bounds(to);
```

Comment:
Is `get_upper_bounds()` part of the final SID runtime API? In the documentation of SID still appears as To Do

Reply:
It is more or less final. I will remove TODO soon. FYI: The API that is in the `concept.hpp` is quite low level and sometimes not that handy to use. The rest of the `sid` directory contains the helpers of various kinds that could be useful.